test/io-tests/test_io.py

```diff
@@ -0,0 +1,129 @@
+'Tests for inputs and outputs of PostgREST.'
+
+import contextlib
+import pathlib
+import subprocess
+import tempfile
+import os
+import time
+
+import pytest
+import requests
+
+BASEURL = 'http://127.0.0.1:49421'
+
+
+basedir = pathlib.Path(os.path.realpath(__file__)).parent
+configs = [path for path in (basedir / 'configs').iterdir() if path.is_file()]
+expectedconfigs = list((basedir / 'configs' / 'expected').iterdir())
+secrets = [path for path in (basedir / 'secrets').iterdir() if path.suffix != '.jwt']
+
+
+@pytest.fixture(params=configs, ids=[conf.name for conf in configs])
+def configpath(request):
+    'Fixture for all config paths.'
+    return basedir / 'configs' / request.param
+
+
+@pytest.fixture(params=expectedconfigs, ids=[conf.name for conf in expectedconfigs])
+def expectedconfig(request):
+    'Fixture for all expected configs.'
+    return request.param
+
+
+@pytest.fixture(params=secrets, ids=[secret.name for secret in secrets])
+def secretpath(request):
+    'Fixture for all secrets.'
+    return request.param
+
+
+def dumpconfig(configpath, moreenv=None):
+    'Dump the config as parsed by PostgREST.'
+    env = os.environ
+    if moreenv:
+        env = {**env, **moreenv}
+
+    command = ['postgrest', '--dump-config', configpath]
+    result = subprocess.run(command, env=env, capture_output=True, check=True)
+    return result.stdout.decode('utf-8')
+
+
+@contextlib.contextmanager
+def run(configpath, stdin=None):
+    'Run PostgREST.'
+    command = ['postgrest', configpath]
+    process = subprocess.Popen(command, stdin=subprocess.PIPE)
+
+    try:
+        if stdin:
+            process.stdin.write(stdin)
+        process.stdin.close()
+
+        waitfor200(BASEURL)
+        yield BASEURL
```

Comment:
If `BASEURL` was created dynamically, we could run the tests in parallel, right? Could speed them up. Irrelevant for CI, because they are quite short, but for `postgrest-watch` it'd be nice.

Reply:
Not sure if pytest can run them in parallel, easily?

Reply:
Yes - we could even run them via temporary sockets. We would need to change all config files and adjust io-tests.sh a bit, so I'm savin that for last :-)

Reply:
Yes, there's a plugin for that - easy!

Reply:
We do have a test for connecting through sockets in place. So if all of the tests are changed to use sockets by default (which I think is good!), we should swap that test around to test connection via localhost.

Reply:
Yes, good point! And that one we can also easily put on a random free port, so no more hardcoded and non-parallelizable endpoints